### ISSUE TITLE
Unhandled exception error when attempting to run No-Bling-builder.sh on macOS Mojave

### ISSUE THREAD

**Issue by Lethalmathematix** [No-Bling/DOTA#50]
It's finally out! 😃 🍻   

&nbsp;

I get this error when running "No-Bling-builder.sh". 

I also get the same error every time I run the compiled vpkmod file without the -s option.

Guess it's something to do with the code to display UI? Because I don't see anything popup where I can modify the default options.

The script seems to work though, and I get the .desktop file, and the compiled vpkmod executable, and an output .vpk file in the tempcontent folder.

I'm not sure if this .vpk file is working though- since I couldn't see the UI, I added "GlanceValue" to "var def" in the .cs file and recompiled it.

Seemed successful: "[new patch] Couriers, Wards, Wearables, Abilities, GlanceValue, Taunts, Loadout"

But running dota with this .vpk file still shows the cosmetics.

&nbsp;

Here's a screenshot of the error I get on running the compiled vpkmod file:

<img width="1440" alt="Screenshot 2020-07-12 at 9 06 52 AM" src="https://user-images.githubusercontent.com/10322629/87238442-324b1380-c420-11ea-8b45-4da21aabd309.png">

&nbsp;

Version of Mono I'm using:

<img width="535" alt="Untitled" src="https://user-images.githubusercontent.com/10322629/87238712-97543880-c423-11ea-984f-83558a5edd85.png">

**Comment by AveYo** [comment]
Yeah I haven't talked about macOS because it sucks.  
First, WinForms do not work on 64bit.  
Then, Catalina removed 32bit support completely.  
Honestly surprised it even worked so far :)  

Don't get your hopes up, but try changing `No-Bling-builder.sh`:   
``` sh
# launch vpkmod with -b option for building No-Bling DOTA mod
mono vpkmod -b
```
to  
``` sh
# launch vpkmod with -b option for building No-Bling DOTA mod
mono --arch=32 vpkmod -b
```

I will probably need to detect macOS and disable choices dialog, and instead use an environment variable

**Comment by Lethalmathematix** [comment]
That works. Thanks! :D

I'm not upgrading to Catalina any time soon. 👀 

Tbh, as long as I can pass the choices via commandline or worst case, set them in the .cs file before compiling, it's fine. As long as the mod works.

**Comment by AveYo** [comment]
@Lethalmathematix that's great!   
I have no way of testing macOS myself, and only added theoretical support but did not know if the STEAMPATH works and file handling in the code does not break at something.   
I guess I'll add this workaround by default on macOS, and also add a line to disable dialog with a note about 64bit.

**Comment by Lethalmathematix** [comment]
Sounds good. :D

Guess I am the macOS guinea pig now. :p

**Comment by dovahkrid** [comment]
waiting for catalina way to work :'(

**Comment by RaphaelDDL** [comment]
I'm also on Mojave and I got the same error as OP. Difference is that when I add the `--arch=32` as @Lethalmathematix mentioned, I get an error instead

```
Error: --arch=32 Failed to switch to 'mono32'.
Press any key to start DOTA
```

Upon further research, I discovered that `brew install mono` does NOT install `mono32` (or at least, it was not adding the bin to `/usr/local/bin/` nor anywhere else so `which mono32` was failing. I uninstalled the brew version and downloaded the pkg from the mono-project website instead. This one installed both mono and mono32, and the .sh opened mono32.

![image](https://user-images.githubusercontent.com/1087109/93152932-594eeb80-f6d6-11ea-8745-3a2e03e23492.png)

So, for the next people on Mac who installed mono through `brew` and had same issues, don't :D Install from https://www.mono-project.com/download/stable/ instead.

On a secondary issue, upon mono32 "booting up", another error is printed in terminal:

```
2020-09-14 22:08:33.545 mono32[9977:1008464] *** Assertion failure in +[NSServiceViewController currentAppIsViewService], /BuildRoot/Library/Caches/com.apple.xbs/Sources/ViewBridge/ViewBridge-401.1/NSViewService.m:131
2020-09-14 22:08:33.545 mono32[9977:1008464] *** Assertion failure in +[NSServiceViewController currentAppIsViewService], /BuildRoot/Library/Caches/com.apple.xbs/Sources/ViewBridge/ViewBridge-401.1/NSViewService.m:131
```

If I try to interact with the options, mono32 hangs. But if I straight out click OK, then it works :D



For a "quick n dirty Mac check" in the script, I've made a PR: https://github.com/No-Bling/DOTA/pull/62/files
